src/os_posix/os_time.c

```diff
@@ -27,10 +28,17 @@ __wt_epoch(WT_SESSION_IMPL *session, struct timespec *tsp)
 	tsp->tv_sec = 0;
 	tsp->tv_nsec = 0;
 
+	/*
+	 * Read into a local variable so that we're comparing the correct
+	 * value when we check for monotonic increasing time.  There are
+	 * many places we read into an unlocked global varible.
+	 */
 #if defined(HAVE_CLOCK_GETTIME)
-	WT_SYSCALL_RETRY(clock_gettime(CLOCK_REALTIME, tsp), ret);
+	WT_SYSCALL_RETRY(clock_gettime(CLOCK_REALTIME, &tmp), ret);
 	if (ret == 0) {
-		__wt_time_check_monotonic(session, tsp);
+		tsp->tv_sec = tmp.tv_sec;
+		tsp->tv_nsec = tmp.tv_nsec;
+		__wt_time_check_monotonic(session, &tmp);
```

Comment:
Yes, good catch.  I have pushed a fix.

I agree that the shared fields are worrying, but I added a WT_DIAGNOSTIC_YIELD call in between the setting of the two fields and the CONFIG ran a long time, until I killed it.